src/hip_memory.cpp

```diff
@@ -1510,6 +1510,15 @@ __global__ void hip_copy2d_n(T* dst, const T* src, size_t width, size_t height,
 }
 }  // namespace
 
+//Checks if operation is inbound, i.e. sizeBytes is no more than allocated size
+inline bool isMemoryOperationInbound(void* dst, size_t sizeBytes) {
+    size_t allocSize = 0;
+    if ((hipMemPtrGetInfo(dst, &allocSize) == hipSuccess) && (allocSize >= sizeBytes)) {
```

Comment:
HIP APIs should not call other HIP APIs. It messes up the HIP trace. You might need to move the implementation of ```hipMemPtrGetInfo``` to an internal function such as ```ihipMemPtrGetInfo``` and then use that both here as well as in ```hipMemPtrGetInfo```.

Reply:
Rewrote without hip trace APIs, please review.